src/stacked_borrows.rs

```diff
@@ -300,10 +300,15 @@ impl<'tcx> Stack {
         item: &Item,
         tag: Option<SbTag>,
         global: &GlobalState,
+        cause: AccessKind,
```

Comment:
Please keep `global` the last argument.

Also please explain in the doc comment that `cause` is for diagnostics only. (And in fact, so is the existing `tag`, which should have been documented but wasn't.)